cinema/views.py

```diff
@@ -0,0 +1,42 @@
+from django.shortcuts import render, get_object_or_404
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework.templatetags.rest_framework import data
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+@api_view(["GET", "POST"])
+def movie_list(request):
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    if request.method == "POST":
+        serializer = MovieSerializer(data, request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, movie_id):
+    movie = get_object_or_404(Movie, pk=movie_id)
+
+    if request.method == "GET":
+        movie_serializer = MovieSerializer(movie)
+        return Response(movie_serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "PUT":
+        movie_serializer = MovieSerializer(instance=movie, data=request.data)
+        if movie_serializer.is_valid():
+            movie_serializer.save()
+            return Response(movie_serializer.data, status=status.HTTP_200_OK)
+        return Response(movie_serializer.errors, status=status.HTTP_400_BAD_REQUEST)

```

Comment:
use is_valid(raise_exception=True) to do all checks automatically and raise handled errors if it's needed